src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -640,25 +644,55 @@ private void ExecuteClusterTransactionOneByOne(List<ClusterTransactionCommand.Si
                 {
                     ClusterWideTransactionIndexWaiter.NotifyListenersAboutError(e);
                     OnClusterTransactionCompletion(command, exception: e);
-                    NotificationCenter.Add(AlertRaised.Create(
-                        Name,
-                        "Cluster transaction failed to execute",
+                    _clusterTransactionErrorNotification.Notify(
                         $"Failed to execute cluster transactions with raft index: {command.Index}. {Environment.NewLine}" +
                         $"With the following document ids involved: {string.Join(", ", command.Commands.Select(item => item.Id))} {Environment.NewLine}" +
-                        "Performing cluster transactions on this database will be stopped until the issue is resolved.",
-                        AlertType.ClusterTransactionFailure,
-                        NotificationSeverity.Error,
-                        $"{Name}/ClusterTransaction",
-                        new ExceptionDetails(e)));
+                        "Performing cluster transactions on this database will be stopped until the issue is resolved.", e);
 
                     DatabaseShutdown.WaitHandle.WaitOne(_clusterTransactionDelayOnFailure);
                     _clusterTransactionDelayOnFailure = Math.Min(_clusterTransactionDelayOnFailure * 2, 15000);
 
                     return;
                 }
+
+                _clusterTransactionErrorNotification.Dismiss();
             }
         }
 
+        private class ClusterTransactionErrorNotification
+        {
+            private readonly NotificationCenter.NotificationCenter _notificationCenter;
+            private readonly string _key;
+            private readonly string _id;
+            private readonly string _databaseName;
+
+            public ClusterTransactionErrorNotification(NotificationCenter.NotificationCenter notificationCenter, string databaseName)
+            {
+                _notificationCenter = notificationCenter;
+                _key = $"{databaseName}/ClusterTransaction";
+                _id = AlertRaised.GetKey(AlertType.ClusterTransactionFailure, _key);
+                _databaseName = databaseName;
+            }
+
+            public void Notify(string msg, Exception e)
+            {
+                _notificationCenter.Add(AlertRaised.Create(
+                    _databaseName,
+                    "Cluster transaction failed to execute",
+                    msg,
+                    AlertType.ClusterTransactionFailure,
+                    NotificationSeverity.Error,
+                    _key,
+                    new ExceptionDetails(e)));
+            }
+
+            public void Dismiss()
+            {
+                if (_notificationCenter.Exists(_id))
```

Comment:
I'm a bit worried here, this opens read tx each time, should we monitor the existence state in memory? You could check if that notification exists e.g. in ctor and then just act accordingly to that state, same with Notify method, that state could be updated afterwards